api/provider/aws/load_balancer_read.go

```diff
@@ -0,0 +1,110 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+// todo: catch 'EntityDoesNotExist' errors
+func (l *LoadBalancerProvider) Read(loadBalancerID string) (*models.LoadBalancer, error) {
+	fqLoadBalancerID := addLayer0Prefix(l.Config.Instance(), loadBalancerID)
+	loadBalancer, err := l.describeLoadBalancer(fqLoadBalancerID)
+	if err != nil {
+		return nil, err
+	}
+
+	ports := make([]models.Port, len(loadBalancer.ListenerDescriptions))
+	for i, description := range loadBalancer.ListenerDescriptions {
+		port := models.Port{
+			ContainerPort: aws.Int64Value(description.Listener.InstancePort),
+			HostPort:      aws.Int64Value(description.Listener.LoadBalancerPort),
+			Protocol:      aws.StringValue(description.Listener.Protocol),
+		}
+
+		if certificateARN := aws.StringValue(description.Listener.SSLCertificateId); certificateARN != "" {
+			// certificate arn format:  arn:aws:iam:region:012345678910:certificate/path/to/name
+			split := strings.SplitN(certificateARN, "/", -1)
+			certificateName := split[len(split)-1]
+			port.CertificateName = certificateName
+		}
+
+		ports[i] = port
+	}
+
+	healthCheck := models.HealthCheck{
+		Target:             aws.StringValue(loadBalancer.HealthCheck.Target),
+		Interval:           int(aws.Int64Value(loadBalancer.HealthCheck.Interval)),
+		Timeout:            int(aws.Int64Value(loadBalancer.HealthCheck.Timeout)),
+		HealthyThreshold:   int(aws.Int64Value(loadBalancer.HealthCheck.HealthyThreshold)),
+		UnhealthyThreshold: int(aws.Int64Value(loadBalancer.HealthCheck.UnhealthyThreshold)),
+	}
+
+	model := &models.LoadBalancer{
+		LoadBalancerID: loadBalancerID,
+		IsPublic:       aws.StringValue(loadBalancer.Scheme) == "internet-facing",
+		URL:            aws.StringValue(loadBalancer.DNSName),
+		Ports:          ports,
+		HealthCheck:    healthCheck,
+	}
+
+	if err := l.readTags(loadBalancerID, model); err != nil {
+		return nil, err
+	}
+
+	return model, nil
+}
+
+func (l *LoadBalancerProvider) describeLoadBalancer(loadBalancerName string) (*elb.LoadBalancerDescription, error) {
+	input := &elb.DescribeLoadBalancersInput{}
+	input.SetLoadBalancerNames([]*string{aws.String(loadBalancerName)})
+	input.SetPageSize(1)
+
+	output, err := l.AWS.ELB.DescribeLoadBalancers(input)
+	if err != nil {
+		return nil, err
+	}
+
+	return output.LoadBalancerDescriptions[0], nil
```

Comment:
Usually this would seem really odd - accessing an array without checking the array's length but i'm guessing if the call to `DescribeLoadBalancers` doesn't return an error, the output will *definitely* contain a single loadbalancer element?